kotlinx-coroutines-core/common/src/channels/Channel.kt

```diff
@@ -105,33 +105,45 @@ public interface SendChannel<in E> {
      * If the channel is closed already, the handler is invoked immediately.
      *
      * The meaning of `cause` that is passed to the handler:
-     * * `null` if the channel was closed or cancelled without the corresponding argument
-     * * the cause of `close` or `cancel` otherwise.
+     * - `null` if the channel was closed normally without the corresponding argument.
+     * - Instance of [CancellationException] if the channel was cancelled normally without the corresponding argument.
+     * - The cause of `close` or `cancel` otherwise.
      *
-     * Example of usage (exception handling is omitted):
+     * ### Execution context and exception safety
      *
+     * The [handler] is executed as part of the closing or cancelling operation, and only after the channel reaches its final state.
+     * This means that if the handler throws an exception or hangs, the channel will still be successfully closed or cancelled.
+     * Unhandled exceptions from [handler] are propagated to the closing or cancelling operation's caller.
+     *
+     * Example of usage:
      * ```
-     * val events = Channel(UNLIMITED)
+     * val events = Channel<Event>(UNLIMITED)
      * callbackBasedApi.registerCallback { event ->
      *   events.trySend(event)
+     *       .onFailure { ... buffer overflown, trySend is unsuccessful ... }
+     *       .onClosed { ... channel is already closed, but callback hasn't stopped yet ... }
```

Comment:
```suggestion
     *       .onFailure { /* buffer overflown, trySend is unsuccessful */ }
     *       .onClosed { /* channel is already closed, but callback hasn't stopped yet */ }
```

Mixing text and code leads to a festive feel, with some punctuation and words in different colors.